File: app/src/main/java/com/regalrooms/tenant/models/Announcement.java
package com.regalrooms.tenant.models;

import com.google.gson.annotations.SerializedName;
import java.util.Date;
import java.util.List;

public class Announcement {
    @SerializedName("_id")
    private String id;
    private String title;
    private String message;
    private String priority;
    private Date createdAt;
    private List<SeenBy> seenBy;

    public static class SeenBy {
        private String tenantId;
        private String name;
        private Date seenAt;

        public String getTenantId() { return tenantId; }
        public String getName() { return name; }
        public Date getSeenAt() { return seenAt; }
    }

    // Getters and Setters
    public String getId() { return id; }
    public String getTitle() { return title; }
    public String getMessage() { return message; }
    public String getPriority() { return priority; }
    public Date getCreatedAt() { return createdAt; }
    public List<SeenBy> getSeenBy() { return seenBy; }
}

File: src/backend/models/Announcement.js
const mongoose = require("mongoose");

const announcementSchema = new mongoose.Schema({

    title:{
        type:String,
        required:true
    },

    message:{
        type:String,
        required:true
    },

    priority:{
        type:String,
        enum:["normal","important","urgent"],
        default:"normal"
    },

    seenBy:[
        {
            tenantId:{
                type:mongoose.Schema.Types.ObjectId,
                ref:"Tenant"
            },

            name:String,

            seenAt:{
                type:Date,
                default:Date.now
            }
        }
    ],

    createdAt:{
        type:Date,
        default:Date.now
    }

});

module.exports = mongoose.model("Announcement",announcementSchema);